### ISSUE TITLE
CloudError: Endpoint (`observeMySession`)

### ISSUE THREAD

**Issue by tlgtrn** [balderdashy/sails#7292]
**Node version**: 20.5.0
**Sails version** _(sails)_: 1.5.7 
**ORM hook version** _(sails-hook-orm)_: 4.0.0 
**Sockets hook version** _(sails-hook-sockets)_: 2.0.0
**Organics hook version** _(sails-hook-organics)_: 2.0.0
**Grunt hook version** _(sails-hook-grunt)_: 5.0
<hr/>

First install according to "get-started" and created "Web App"

Signed up with email address and when I try login I am facing this error.

"CloudError: Endpoint (`observeMySession`) responded with an error (or the request failed)."

Uncaught (in promise) CloudError: Endpoint (`observeMySession`) responded with an error (or the request failed).
    exec http://localhost:1337/dependencies/cloud.js:649
    _promise http://localhost:1337/dependencies/cloud.js:612
    toPromise http://localhost:1337/dependencies/cloud.js:610
    then http://localhost:1337/dependencies/cloud.js:592

cloud.js:649:33

**Comment by usmanghani599** [comment]
I am also facing same issue. Any solution please?

**Comment by sreehari-nallapaneni** [comment]
Hi, As a temporary fix you can try the following.

step 1: Open `your-project\assets\js\components\account-notification-banner.component.js`
step 2: Replace **mounted** method with following
  ```
mounted: async function() {
    try{
      await Cloud.observeMySession();
      // Listen for updates to the user's session
      Cloud.on('session', (msg)=>{
        if(msg.notificationText) {
          this.notificationText = msg.notificationText;
        } else {
          this.notificationText = '';
        }
      });//œ
    }
    catch(e){
      console.log(`some error : `, e);
    }
  }
```

Reason replacing **mounted** method is, in the **mounted** method `await Cloud.observeMySession();` throws an error that is not handled, So we are handling that error with try/catch.

**Comment by rizikoblogger** [comment]
I am facing the same.

The temporary solution above works, but we keep without WebSocket capability because 'sails.sockets' is working nomore.

It looks like _csrf is not being sent into window.SAILS_LOCALS, so

I recommend you set false csrf attribute into config/security.js file - for a while.

I mean - FOR A WHILE - due to security issues around lack of CSRF control.

**Comment by mikermcneil** [comment]
@rizikoblogger @sreehari-nallapaneni @usmanghani599 @tlgtrn Patch otw (see also #7297)

**Comment by mikermcneil** [comment]
Published as v1.5.8
